tastypie/authorization.py

```diff
@@ -263,3 +257,88 @@ def delete_detail(self, object_list, bundle):
             raise Unauthorized("You are not allowed to access that resource.")
 
         return True
+
+
+class DjangoObjectAuthorization(Authorization):
+    '''
+    Uses permission checking from ``django.contrib.auth`` to map
+    ``POST / PUT / DELETE / PATCH`` to their equivalent Django auth
+    permissions.
+
+    Both the list & detail simply check the object they're based on.
+    Object level authorization api is added since django 1.5. However,
+    it will default to no-access unless an AUTHENTICATION_BACKENDS is
+    setup to handle those checks.
+    '''
+
+    # By default, follows `ModelAdmin` "convention" to use `app.change_model`
+    # `django.contrib.auth.models.Permission` for both viewing and updating.
+    # https://docs.djangoproject.com/es/1.9/topics/auth/default/#permissions-and-authorization
+    READ_PERM_CODE = getattr(settings, 'TASTYPIE_READ_PERM_CODE', 'change')
```

Comment:
What you proposed makes sense. I will add it a patch (and `settings` to django...)

If you think it makes sense to include `DjangoObjectAuthorization`, would you like me to collapse the two classes. The `DjangoObjectAuthorization` and `DjangoAuthorization` really only different in a single method call, namely `user.has_perm(permission, obj_or_list)` vs `user.has_perm(permission)`.

Something like the code below. 

(I am in the process of updating my own app to Django 1.8 (yeah, a bit late to the party) and in a bit of a git mess. Will update the pull request once I get back to the stage that I can run test. Sorry that I might not able to update this PR until sometime next week. Sorry about that)

```python
class DjangoAuthorization(Authorization):
    """
    Uses permission checking from ``django.contrib.auth`` to map
    ``POST / PUT / DELETE / PATCH`` to their equivalent Django auth
    permissions.

    Both the list & detail variants simply check the model they're based
    on, as that's all the more granular Django's permission setup gets.
    """

    # By default, following `ModelAdmin` "convention", `app.change_model` is used
    # `django.contrib.auth.models.Permission` as perm code for viewing and updating.
    # https://docs.djangoproject.com/es/1.9/topics/auth/default/#permissions-and-authorization
    READ_PERM_CODE = getattr(settings, 'TASTYPIE_READ_PERM_CODE', 'change')

    def base_checks(self, request, model_klass):
        # If it doesn't look like a model, we can't check permissions.
        if not model_klass or not getattr(model_klass, '_meta', None):
            return False

        # User must be logged in to check permissions.
        if not hasattr(request, 'user'):
            return False

        return model_klass

    def check_user_perm(self, user, permission, obj_or_list):
        return user.has_perm(permission)

    def perm_list_checks(self, request, code, obj_list):
        klass = self.base_checks(request, obj_list.model)
        if klass is False:
            return []

        permission = '%s.%s_%s' % (
            klass._meta.app_label,
            code,
            get_module_name(klass._meta)
        )

        if self.check_user_perm(request.user, permission, obj_list):
            return obj_list

        return obj_list.none()

    def perm_obj_checks(self, request, code, obj):
        klass = self.base_checks(request, obj.__class__)
        if klass is False:
            raise Unauthorized("You are not allowed to access that resource.")

        permission = '%s.%s_%s' % (
            klass._meta.app_label,
            code,
            get_module_name(klass._meta)
        )

        if request.user.has_perm(permission, obj):
            return True

        return False

    def read_list(self, object_list, bundle):
        return self.perm_list_checks(bundle.request, self.READ_PERM_CODE, object_list)

    def read_detail(self, object_list, bundle):
        return self.perm_obj_checks(bundle.request, self.READ_PERM_CODE, bundle.obj)

    def create_list(self, object_list, bundle):
        return self.perm_list_checks(bundle.request, 'add', object_list)

    def create_detail(self, object_list, bundle):
        return self.perm_obj_checks(bundle.request, 'add', bundle.obj)

    def update_list(self, object_list, bundle):
        return self.perm_list_checks(bundle.request, 'change', object_list)

    def update_detail(self, object_list, bundle):
        return self.perm_obj_checks(bundle.request, 'change', bundle.obj)

    def delete_list(self, object_list, bundle):
        return self.perm_list_checks(bundle.request, 'delete', object_list)

    def delete_detail(self, object_list, bundle):
        return self.perm_obj_checks(bundle.request, 'delete', bundle.obj)

class DjangoObjectAuthorization(DjangoAuthorization):
    '''
    Uses permission checking from ``django.contrib.auth`` to map
    ``POST / PUT / DELETE / PATCH`` to their equivalent Django auth
    permissions.

    Both the list & detail simply check the object they're based on.
    Object level authorization api is added since django 1.5. However,
    it will default to no-access unless an AUTHENTICATION_BACKENDS is
    setup to handle those checks.
    '''

    def check_user_perm(self, user, permission, obj_or_list):
        return user.has_perm(permission, obj_or_list)
```